logstash-core/src/main/java/org/logstash/FileLockFactory.java

```diff
@@ -39,7 +41,8 @@
 
     private FileLockFactory() {}
 
-    private static final Set<String> LOCK_HELD = Collections.synchronizedSet(new HashSet<String>());
+    private static final Set<String> LOCK_HELD = Collections.synchronizedSet(new HashSet<>());
```

Comment:
pretty simple: you want to allow releasing a lock if and only if it was first obtained, **thought the `FileLockFactory`**  we don't support any other locking mecanism, it has to be obtained using this class.

Sure, we could rename it to FileLockManager, but I would defer that at this point?